aws/resource_aws_elasticsearch_domain.go

```diff
@@ -472,6 +473,21 @@ func resourceAwsElasticSearchDomainCreate(d *schema.ResourceData, meta interface
 			if isAWSErr(err, "ValidationException", "The passed role has not propagated yet") {
 				return resource.RetryableError(err)
 			}
+			if isAWSErr(err, "ValidationException", "Authentication error") ||
+				isAWSErr(err, "ValidationException", "Unauthorized Operation: Elasticsearch must be authorised to describeVpcs") ||
+				isAWSErr(err, "ValidationException", "Unauthorized Operation: Elasticsearch must be authorised to describeSubnets") ||
+				isAWSErr(err, "ValidationException", "Before you can proceed, you must enable a service-linked role to give Amazon ES permissions to access your VPC") {
+				log.Printf("[DEBUG] Retrying creation of ElasticSearch domain %s (transient error)", aws.StringValue(input.DomainName))
+				return resource.RetryableError(err)
+			}
```

Comment:
Can you please split these into separate conditionals? The third parameter is also uses `strings.Contains()`, so messages can be consolidated as well. The `enable a service-linked role` error is already handled on line 463.

```go
			if isAWSErr(err, "ValidationException", "Authentication error") {
				return resource.RetryableError(err)
			}
			// Example: ValidationException: Unauthorized Operation: Elasticsearch must be authorised to describeSubnets
			// Example: ValidationException: Unauthorized Operation: Elasticsearch must be authorised to describeVpcs
			if isAWSErr(err, "ValidationException", "Unauthorized Operation: Elasticsearch must be authorised")  {
				return resource.RetryableError(err)
			}
```

We also have an additional error condition to retry on if you could add it as well please:

```
--- FAIL: TestAccAWSElasticSearchDomain_CognitoOptionsCreateAndRemove (17.30s)
    testing.go:635: Step 0 error: errors during apply:
        
        Error: Error creating ElasticSearch domain: ValidationException: The passed role must authorize Amazon Elasticsearch to describeUserPool
```

```go
			// Example: ValidationException: The passed role must authorize Amazon Elasticsearch to describeUserPool
			if isAWSErr(err, "ValidationException", "The passed role must authorize") {
				return resource.RetryableError(err)
			}
```

It is a shame the Elasticsearch API does not use a separate error code to signal IAM issues instead. 😅

Reply:
@bflad currently implementing the requested changes. Many thanks for this review
BTW if you or someone could also have a look at #11671 that would be awesome